python/frequenz/microgrid_component_graph/__init__.pyi

```diff
@@ -0,0 +1,334 @@
+# License: MIT
+# Copyright © 2025 Frequenz Energy-as-a-Service GmbH
+
+# Type hints for the component graph module.
+
+"""A graph representation of the electrical components in a microgrid."""
+
+from collections import abc
```

Comment:
I think we should use `from collections.abc import Iterable, Set`. IIRC either `pydoclint` or `mkdocstrings` worked better when these basic types are imported to the top-level.